vou/person.py

```diff
@@ -220,25 +227,117 @@ def compute_OD_dose_reduction(self):
         else:
             return dose_reduction
 
+   
+
     def will_increase_dose(
-        self, effect_window: int = 20, increase_threshold: float = 0.4,
+        self, t:int, effect_window: int = 20, increase_threshold: float = 0.4,
     ):
         """
         Checks whether the person will increase their dose. Based on a comparison of
         the average of past dose effects to the person's desired dose. Effect window
         and increase threshold are calibrated parameters and not intended to be varied
         during simulation.
+
+        Logic for the source of the increased dose is added here.  If a person will increase
+        their dose for the first time, they will attempt through their primary doctor.  If
+        this is their second time, they will have a 50% chance of trying through a secondary
+        doctor or a dealer.
         """
+        
+        if len(list(self.dose_increase_record.values())) == 0:
+            last_attempt_src = "primary_doctor"
```

Comment:
I don't understand why we are assigning their last attempt as primary doctor if this is their first attempt.